reactioncommerce/reaction
# (fix): fix permissions of shop social settings

Resolves #4205 
Impact: major
Type: bugfix

## Issue
`socialPackage` returned in `socialSettingsContainer` is always the package from the primary shop. This causes the social settings to be displayed to the dashboard of other shop owners.

In displaying the social buttons in product page, the settings of the primary shop owners are always used. Example, if product is from primary shop where all social settings are enabled, then all social buttons should be displayed. If product is from another shop where only Twitter is enabled, then product page should only show the Twitter button, but this is currently not the case.

## Solution
Add `shopId` filter to `socialSettingsContainer` and `createSocialSettings`. 

## Breaking changes
None


## Testing
1. Login as admin. Set the social settings in primary shop (all enabled). Publish a product.
2. Enable the marketplace. Invite a shop owner and activate the shop.
3. Login as shop owner. Set the shop's social settings (some possibly disabled). You should be able to see and save only the shop's credentials.
